src/engraving/dom/chord.cpp

```diff
@@ -2774,6 +2774,29 @@ EngravingItem* Chord::prevElement()
     }
     switch (e->type()) {
     case ElementType::NOTE: {
+        if (isGrace()) {
+            ChordRest* next = prevChordRest(this);
+            if (next) {
+                return toChord(next)->notes().back();
+            }
+        }
+
+        if (!graceNotesBefore().empty()) {
```

Comment:
It would be better to call graceNotesBefore()/graceNotesAfter() only once. Also, ideally, we should check if notes() is empty before calling back()